### DISCUSSION TITLE
Image from CDN via Link, NOT downloaded in build ?

### DISCUSSION THREAD

**Discussion by tob-har** [squidfunk/mkdocs-material#7323] [Q&A] [answered]
I need images embedded with a link from a CDN.
Is it possible to build the site, without downloading the files and putting them into a local asset, but keeping
the links to the CDN instead?

**Comment by alexvoss** [comment]
According to the [Markdown documentation](https://daringfireball.net/projects/markdown/syntax#img) you should be able to just specify a URL to the image in the Markdown instead of a local path.

**Reply by alexvoss** [reply]
For example, this works`![Alex Voss](https://avatars.githubusercontent.com/u/4134224?v=4)`

**Reply by tob-har** [reply]
yes, sure, that's what I am doing. But in the build process, the image is downloaded and put into a local asset folder. That makes sense of course for many reasons, but I need to avoid this behavior and keep source code referencing the original link to the CDN and grab the source file from there...

**Reply by alexvoss** [reply, marked as answer]
Do you have the [privacy plugin](https://squidfunk.github.io/mkdocs-material/plugins/privacy/) active by any chance? That would be the bit downloading the images.

**Reply by tob-har** [reply]
yes... thanks for this hint. Will give that a try.

**Reply by tob-har** [reply]
Thanks. That's the solution.

**Reply by squidfunk** [reply]
Note that with [Insiders](https://squidfunk.github.io/mkdocs-material/insiders/), you can exclude certain domains or URL patterns from being downloaded via [`assets_exclude`](https://squidfunk.github.io/mkdocs-material/plugins/privacy/#config.assets_exclude):

``` yaml
plugins:
  - privacy:
      assets_exclude: 
        - unpkg.com/mathjax@3/*
        - giscus.app/*
```
